dwave/optimization/include/dwave-optimization/nodes/numbers.hpp

```diff
@@ -67,90 +69,171 @@ class NumberNode : public ArrayOutputMixin<ArrayNode>, public DecisionNode {
     template <std::uniform_random_bit_generator Generator>
     void initialize_state(State& state, Generator& rng) const {
         std::vector<double> values;
-
         const ssize_t size = this->size();
         values.reserve(size);
 
         if (integral()) {
-            std::uniform_int_distribution<ssize_t> gen(lower_bound_, upper_bound_);
-            for (ssize_t i = 0; i < size; ++i) values.emplace_back(gen(rng));
+            for (ssize_t i = 0; i < size; ++i) {
+                std::uniform_int_distribution<ssize_t> gen(lower_bound(i), upper_bound(i));
+                values.emplace_back(gen(rng));
+            }
         } else {
-            std::uniform_real_distribution<double> gen(lower_bound_, upper_bound_);
-            for (ssize_t i = 0; i < size; ++i) values.emplace_back(gen(rng));
+            for (ssize_t i = 0; i < size; ++i) {
+                std::uniform_real_distribution<double> gen(lower_bound(i), upper_bound(i));
+                values.emplace_back(gen(rng));
+            }
         }
-
         return initialize_state(state, std::move(values));
     }
 
-    // Specializations for the linear case
+    // NumberNode methods *****************************************************
+
+    // In the given state, swap the value of index i with the value of index j.
+    // Returns `true` if the values at indices i and j change and `false`
+    // otherwise.
     bool exchange(State& state, ssize_t i, ssize_t j) const;
+
+    // Return the value of index i in a given state.
     double get_value(State& state, ssize_t i) const;
 
+    // Specialization of the linear case
```

Comment:
Not sure what this comment means. But we should consider deleting this method, I don't think it belongs here